### PR TITLE
Workflow context

### LINKED ISSUE #39: API to access Github Event Payload
Currently, is not possible to access `github.event` object using this library. I propose building a similar interface as the official [actions toolkit](https://github.com/actions/toolkit/blob/main/packages/github/src/context.ts#L30).

### PR DESCRIPTION
**Pull request by bernardoVale**
closes #39. @sethvargo  let me know what you think first and I'll write the tests.

### CODE REVIEW

File: actions.go [hunk 1]
```diff
@@ -415,3 +416,77 @@ func (c *Action) Getenv(key string) string {
...
+	Job        string
+	RunNumber  int
+	RunID      int
+	APIURL     string
+	ServerURL  string
+	GraphqlURL string
+}
+
+// Context returns the context of current action with the payload object that triggered the workflow
+func (c *Action) Context() (*Context, error) {
```
Comment (sethvargo) [inline actions.go:438]: I'm generally a fan of keeping dependencies minimal, but have you see https://github.com/sethvargo/go-envconfig? I think that could make this a lot simpler.

Comment (bernardoVale) [reply actions.go:438]: If you think its ok to add this dep, yes it will make things simper and cleaner

Comment (sethvargo) [reply actions.go:438]: Yea I think it makes things a bit cleaner and more flexible.

Comment (bernardoVale) [reply actions.go]: done

File: actions.go [hunk 2]
```diff
@@ -419,31 +419,34 @@ type GetenvFunc func(key string) string
...
 
 // Context returns the context of current action with the payload object that triggered the workflow
-func (c *Action) Context() (*Context, error) {
-	var rawEvent map[string]interface{}
-
-	eventPath := c.Getenv("GITHUB_EVENT_PATH")
-	if eventPath == "" {
-		return nil, fmt.Errorf("missing GITHUB_EVENT_PATH in environment")
+func (c *Action) Context(ctx context.Context) (*Context, error) {
+	var (
```
Comment (sethvargo) [inline actions.go:440]: Do we still need this?

File: actions.go [hunk 3]
```diff
@@ -419,31 +419,34 @@ type GetenvFunc func(key string) string
...
+	EventName        string
+	SHA              string `env:"GITHUB_SHA"`
+	Ref              string `env:"GITHUB_REF"`
+	Workflow         string `env:"GITHUB_WORKFLOW"`
+	Action           string `env:"GITHUB_ACTION"`
+	Actor            string `env:"GITHUB_ACTOR"`
+	Job              string `env:"GITHUB_JOB"`
+	RunNumber        int    `env:"GITHUB_RUN_NUMBER"`
+	RunID            int    `env:"GITHUB_RUN_ID"`
+	APIURL           string `env:"GITHUB_API_URL,default=https://api.github.com"`
```
Comment (sethvargo) [inline actions.go:433]: Is there a reason to have defaults? I feel like, if the environment doesn't inject these values, we shouldn't either. What does the Node SDK do here?

Comment (bernardoVale) [reply actions.go:433]: We're replacating more or less the same logic: https://github.com/actions/toolkit/blob/main/packages/github/src/context.ts#L49-L52

File: actions.go (hunk 2 above)
Comment (bernardoVale) [reply actions.go:440]: not sure if I get it, the variables? Yes

**Review by sethvargo — COMMENTED** [review]
Let me know if you want some guidance on writing tests. It should be a little easier now with envconfig.

File: actions.go (hunk 3 above)
Comment (bernardoVale) [reply actions.go:433]: sure

File: actions.go (hunk 2 above)
Comment (bernardoVale) [reply actions.go:440]: don't worry I can clean that for you

**Pull request closed without merging**